### ISSUE TITLE
POSTing document w/ Mongo ObjectId returns id: [object Object]

### ISSUE THREAD

**Issue by HolySamosa** [balderdashy/sails#2333]
For my application, I need to create my model instances with Mongo ObjectIds generated on the client.
 
Using the provided REST blueprints, I can POST a document provides an 'id' field and it is correctly inserted into the database with 'id' typed as ObjectId.  Future queries will return the correct 'id' value.

Unfortunately, the result of the POST returns a JSON with `"id": "[object Object]"`, which is presenting problems for the client.

Example:

    POST /person
    {
    "id": "544974f107d766a038f6a65e"
  	"name": "george",
    }

    RETURNS:
    {
    "id": "[object Object]"
  	"name": "george",
    }

It's not clear to me if this is a sails issue, or sails-mongo.  

If a fix isn't likely to appear shortly, a recommended work-around would be awesome.  Thanks!

**Comment by sailsbot** [comment]
Thanks for posting, @HolySamosa.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
